### PR TITLE
Fix bug where school doesn't show up in PD auto emails. 

### PR DESCRIPTION
**Pull request by aoby**
Add helper methods and tests.

[Bug](https://trello.com/c/0U3JoExf/123-bug-fix-school-info-in-workshop-auto-emails)

Now, the [mail preview](https://test-studio.code.org/rails/mailers/pd/workshop_mailer/organizer_cancel_receipt) looks like this:
image

### CODE REVIEW

**Comment by aoby** [comment]
In the process I cleaned up some of the enrollment validations, moving toward allowing old enrollments with old rules to be grandfathered in and validating new ones more strictly.

I also added a default `school_info` factory.

File: dashboard/test/controllers/api/v1/pd/workshop_enrollments_controller_test.rb [hunk 1]
```diff
@@ -6,7 +6,6 @@ class Api::V1::Pd::WorkshopEnrollmentsControllerTest < ::ActionController::TestC
     @facilitator = create :facilitator
 
     @workshop = create :pd_workshop, organizer: @organizer, facilitators: [@facilitator]
-    @school_info = create :school_info_without_country
```
Comment (aoby) [inline dashboard/test/controllers/api/v1/pd/workshop_enrollments_controller_test.rb:9]: this was unused

File: dashboard/app/models/pd/enrollment.rb [hunk 2]
```diff
@@ -222,6 +221,19 @@ def self.get_safe_names
     all.map {|enrollment| [enrollment.full_name, enrollment]}
   end
 
+  # TODO: Delete school column
+  def school
+    ActiveSupport::Deprecation.warn('School is deprecated. Use school_info or school_name instead.')
```
Comment (mehalshah) [inline dashboard/app/models/pd/enrollment.rb]: You should still return school here for now

File: dashboard/app/models/pd/enrollment.rb [hunk 3]
```diff
@@ -222,6 +221,19 @@ def self.get_safe_names
     all.map {|enrollment| [enrollment.full_name, enrollment]}
   end
 
+  # TODO: Delete school column
```
Comment (ewjordan) [inline dashboard/app/models/pd/enrollment.rb]: We just need to make sure that getting rid of the column waits on making use of the existing school data (which is on my plate soon)

**Pull request merged**